cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/RemoteConfigurator.java

```diff
@@ -0,0 +1,105 @@
+import io.cdap.common.http.HttpResponse;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * RemoteConfigurator sends a request to another {@link Configurator} running remotely.
+ */
+public class RemoteConfigurator implements Configurator {
+  private static final Logger LOG = LoggerFactory.getLogger(RemoteConfigurator.class);
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(CConfiguration.class, new CConfigurationCodec()).create();
+  private final ConfiguratorConfig config;
+
+  private final RemoteClient remoteClientInternal;
+
+  public RemoteConfigurator(ConfiguratorConfig config,
+                            DiscoveryServiceClient discoveryServiceClient) {
+    this.config = config;
+    this.remoteClientInternal = new RemoteClient(discoveryServiceClient, Constants.Service.TASK_WORKER,
+                                                 new DefaultHttpRequestConfig(false),
+                                                 String.format("%s", Constants.Gateway.INTERNAL_API_VERSION_3));
+  }
+
+  public RemoteConfigurator(CConfiguration cConf, NamespaceId appNamespace, ArtifactId artifactId,
+                            String appClassName, PluginFinder pluginFinder, ClassLoader artifactClassLoader,
+                            String applicationName, String applicationVersion, String configString,
+                            DiscoveryServiceClient discoveryServiceClient,
+                            Location artifactLocation) {
+    this(new ConfiguratorConfig(cConf, appNamespace,
+                                artifactId, appClassName,
+                                applicationName,
+                                applicationVersion,
+                                configString, artifactLocation.toURI()),
+         discoveryServiceClient);
+  }
+
+  @Override
+  public ListenableFuture<ConfigResponse> config() {
+
+    String jsonResponse = "";
+
+    try {
+      String param = GSON.toJson(config);
+      RunnableTaskRequest req = new RunnableTaskRequest(ConfiguratorTask.class.getName(), param);
+      String reqBody = GSON.toJson(req);
+
+      HttpRequest.Builder requestBuilder = remoteClientInternal.requestBuilder(
+        HttpMethod.POST, String.format("/worker/run")).withBody(reqBody);
+      HttpResponse response = remoteClientInternal.execute(requestBuilder.build());
+
+      jsonResponse = response.getResponseBodyAsString();
+    } catch (Exception ex) {
+      LOG.warn(String.format("Exception caught during RemoteConfigurator.config, got json response: %s", jsonResponse),
+               ex);
+      return Futures.immediateFailedFuture(ex);
+    }
+
+    ConfigResponseResult configResponse = GSON.fromJson(jsonResponse, ConfigResponseResult.class);
+    if (configResponse.getException() != null) {
+      //TODO: Refactor to use RemoteTaskExecutor once PR #13383 is merged
```

Comment:
This class will be used to construct the request parameters and then it will call `RemoteTaskExecutor` to handle the request, retries, error handling etc.